parser.cpp

```diff
@@ -30,21 +29,81 @@ Result<std::unique_ptr<ast::Str>> Parser::parseStr() {
 }
 
 Result<std::unique_ptr<ast::Out>> Parser::parseOut() {
+  Result<std::unique_ptr<ast::Str>> sres = parseStr();
+  if (sres.hasError()) return sres;
+  std::unique_ptr<ast::Str> &str = sres.get();
+  return Result<std::unique_ptr<ast::Out>>(
+      std::make_unique<ast::Out>(std::move(str)));
+}
+
+Result<std::unique_ptr<ast::Expr>> Parser::parseExpr() {
   Result<Token> res = lexer_.Lex();
   if (res.hasError()) return res;
+  Token &tok = res.get();
+  if (tok.getKind() == Token::out) {
+    Result<std::unique_ptr<ast::Out>> out = parseOut();
+    if (out.hasError()) return out;
+    return Result<std::unique_ptr<ast::Expr>>(std::move(out.get()));
+  }
 
-  const Token &tok = res.get();
-  if (tok.getKind() != Token::out) {
-    return Result<std::unique_ptr<ast::Out>>::BuildError()
-           << "Expected 'out' on row " << tok.getRow() << ", col "
+  return Result<std::unique_ptr<ast::Expr>>::BuildError()
+         << "Expected '(' on row " << tok.getRow() << ", col " << tok.getCol()
+         << "!!!";
+  ;  // TODO(dsLeks): Implement this.
+}
+
+Result<std::unique_ptr<ast::Prototype>> Parser::parsePrototype(
+    std::string name) {
+  Result<Token> res = lexer_.Lex();
+  if (res.hasError()) return res;
+  Token tok = res.get();
+  if (tok.getKind() != Token::lparen) {
+    return Result<std::unique_ptr<ast::Prototype>>::BuildError()
+           << "Expected '(' on row " << tok.getRow() << ", col "
+           << tok.getCol();
+  }
+  // PARSE RETURN TYPE HERE
```

Comment:
Let's remove these comments or at least make them less loud and sentences: `Parse return type here.`